reana_server/complexity.py

```diff
@@ -309,25 +313,41 @@ def _populate_dependencies(self, steps):
         """Populate dependencies to parsed CWL workflow tree steps."""
         for step, value in steps.items():
             nested_workflow = value.get("workflow")
-            if nested_workflow:
+            if nested_workflow and isinstance(nested_workflow, str):
                 for nested_step, nested_value in steps.items():
                     if nested_workflow in nested_step:
                         nested_value["dependencies"] += value["dependencies"]
         return steps
 
     def _populate_complexity(self, steps):
         """Populate complexity to parsed CWL workflow tree steps."""
-        for step, value in steps.items():
-            scatter_params = value.get("scatter_params")
-            if scatter_params:
-                param_len = len(self.input_params.get(scatter_params, []))
-                if not param_len:
-                    continue
+
+        def _parse_steps(steps):
```

Comment:
Minor: Maybe we can make this a separate method instead of nested one.

Reply:
It was made as a nested function because it is being called recursively. Same technique used for Yadage, so I would propose to leave it as is